client/app/visualizations/pivot/index.js

```diff
@@ -12,48 +15,65 @@ function pivotTableRenderer() {
     template: '',
     replace: false,
     link($scope, element) {
-      $scope.$watch('queryResult && queryResult.getData()', (data) => {
-        if (!data) {
-          return;
-        }
+      function updatePivot() {
+        $scope.$watch('queryResult && queryResult.getData()', (data) => {
+          if (!data) {
+            return;
+          }
 
-        if ($scope.queryResult.getData() !== null) {
-          // We need to give the pivot table its own copy of the data, because it changes
-          // it which interferes with other visualizations.
-          data = $.extend(true, [], $scope.queryResult.getRawData());
-          const options = {
-            renderers: $.pivotUtilities.renderers,
-            onRefresh(config) {
-              const configCopy = Object.assign({}, config);
-              // delete some values which are functions
-              delete configCopy.aggregators;
-              delete configCopy.renderers;
-              delete configCopy.onRefresh;
-              // delete some bulky default values
-              delete configCopy.rendererOptions;
-              delete configCopy.localeStrings;
+          if ($scope.queryResult.getData() !== null) {
+            // We need to give the pivot table its own copy of the data, because it changes
+            // it which interferes with other visualizations.
+            data = $.extend(true, [], $scope.queryResult.getRawData());
+            const options = {
+              renderers: $.pivotUtilities.renderers,
+              onRefresh(config) {
+                const configCopy = Object.assign({}, config);
+                // delete some values which are functions
+                delete configCopy.aggregators;
+                delete configCopy.renderers;
+                delete configCopy.onRefresh;
+                // delete some bulky default values
+                delete configCopy.rendererOptions;
+                delete configCopy.localeStrings;
 
-              if ($scope.visualization) {
-                $scope.visualization.options = configCopy;
-              }
-            },
-          };
+                if ($scope.visualization) {
+                  $scope.visualization.options = configCopy;
+                }
+              },
+            };
 
-          if ($scope.visualization) {
-            Object.assign(options, $scope.visualization.options);
+            if ($scope.visualization) {
+              Object.assign(options, $scope.visualization.options);
+            }
+            $(element).pivotUI(data, options, true);
+            if ($scope.visualization.options.controls.enabled) {
+              const controls = $('.pvtAxisContainer, .pvtRenderer, .pvtVals');
+              for (let i = 0; i < controls.length; i += 1) { controls[i].style.display = 'none'; }
```

Comment:
With .pivot() way, if a user creates pivot table with Heatmap renderer, and then hides the controls, the table will be rendered as a plain white pivot table. That's what .pivot() is. So it does more than just hiding the controls. There is no way to keep the table cell background colors.

My way is a little hackish, but retains the Heatmapped table.